exporter/tanzuobservabilityexporter/exporter.go

```diff
@@ -64,26 +63,23 @@ type tracesExporter struct {
 }
 
 func newTracesExporter(settings component.ExporterCreateSettings, c config.Exporter) (*tracesExporter, error) {
-	cfg, ok := c.(*Config)
+	cfgPtr, ok := c.(*Config)
 	if !ok {
 		return nil, fmt.Errorf("invalid config: %#v", c)
 	}
-
-	endpoint, err := url.Parse(cfg.Traces.Endpoint)
-	if err != nil {
-		return nil, fmt.Errorf("failed to parse traces.endpoint: %v", err)
-	}
-	tracingPort, err := strconv.Atoi(endpoint.Port())
+	cfg, tracesURL, metricsURL, err := cfgPtr.withoutMissingFields()
```

Comment:
The missing URLs do get filled in the returned config struct.  In fact, you could run url.Parse(cfg.Traces.Endpoint). However, withoutMissingFields() has to call url.Parse already to do its job of filling in missing fields in the struct.  Rather than make the caller parse the URL strings again, why not just return the url.URL values?  If you prefer, I can change withoutMissingFields() to return just the config struct and the error, but keep in mind that if we do that, the caller will have to parse the URL strings, check for errors, etc.